scripts/sync_go_toolchain_directive.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+# This script looks at the version present in the .go-version file and treats
+# that to be the value of the toolchain directive that go should use. It then
+# updates the toolchain directives of all go.mod files to reflect this version.
+#
+# We do this to ensure that .go-version acts as the source of truth for go versions.
+
+set -euo pipefail
+
+source ./scripts/test_lib.sh
+
+TARGET_GO_VERSION="${TARGET_GO_VERSION:-"$(cat "${ETCD_ROOT_DIR}/.go-version")"}"
+find . -name 'go.mod' -exec go mod edit -toolchain=go${TARGET_GO_VERSION} {} \;
```

Comment:
There are three options,
- only sync go line, e.g. `go 1.22.2`;
- only sync toolchain line, e.g. `toolchain go1.22.2`;
- sync both go and toolchain lines.

No matter which way we select, it will always use the highest go version by default.

For simplicity, can we go with option 1?

Reply:
@ahrtr, if I understand correctly, the issue I see with option 1 and 3 is that whenever we allow bumping the `go` directive, we are also forcing developers of etcd and other projects that import etcd modules to bump their versions. From https://go.dev/doc/toolchain:
> the go line sets the minimum required Go version necessary to use a module or workspace.

Reply:
Yes, it's true. It's OK to go for option 2. But please also see my next related comment https://github.com/etcd-io/etcd/pull/17876#discussion_r1582930923